tests/Functional/CorporateFooterRenderTest.php

```diff
@@ -0,0 +1,435 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\Functional;
+
+use Drupal\Tests\BrowserTestBase;
+use Symfony\Component\Yaml\Yaml;
+use Drupal\Component\Render\FormattableMarkup;
+use Behat\Mink\Element\NodeElement;
+
+/**
+ * Test footer block rendering.
+ */
+class CorporateFooterRenderTest extends BrowserTestBase {
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'block',
+    'page_header_metadata_test',
+    'oe_theme_helper',
+    'oe_corporate_blocks',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Enable and set OpenEuropa Theme as default.
+    $this->container->get('theme_installer')->install(['oe_theme']);
+    $this->container->get('theme_handler')->setDefault('oe_theme');
+    $this->container->set('theme.registry', NULL);
+
+    // Rebuild the ui_pattern definitions to collect the ones provided by
+    // oe_theme itself.
+    $this->container->get('plugin.manager.ui_patterns')->clearCachedDefinitions();
+    $this->configFactory = $this->container->get('config.factory');
+  }
+
+  /**
+   * Test European Commission footer core block rendering.
+   */
+  public function testEcFooterCoreBlockRendering(): void {
```

Comment:
If we start a new test every time then the cache will not be populated when visiting the page for the first time, preventing us from testing proper invalidation.

Let's have one, big test, instead of 4 separate once. This will allow us to test cache invalidation when we switch from the installation default (ec/core) to ec/standardised, eu/core and eu/standardised: we will just change configuration, reload the page, and the new footer should be there.

Reply:
done, implied a bit of refactoring